### DISCUSSION TITLE
[Help] RampUp - how to increase linearly the amount of users

### DISCUSSION THREAD

**Discussion by joaormouta** [artilleryio/artillery#1763] [Artillery]
Hi,
I have this following scenario:
- duration: 10
  arrivalRate: 1
- duration: 10
  arrivalRate: 1
- duration: 10
  arrivalRate: 1
- duration: 10
  arrivalRate: 2
- duration: 10
  arrivalRate: 2
- duration: 10
  arrivalRate: 2
- duration:10
  arrivalRate: 3
- duration: 10
  arrivalRate: 3
- duration: 10
  arrivalRate: 3
- duration: 10
  arrivalRate: 4
- duration: 10
  arrivalRate: 4
- duration: 10
  arrivalRate: 4
- duration: 10
  arrivalRate: 5
- duration: 10
  arrivalRate: 5
- duration: 10
  arrivalRate: 5
...
Is there a way to simplify this ? ( every 30 seconds i increase a Vuser that works 3 times during 10 seconds)

i have try the rampup but without sucess
- duration: 150
  arrivalRate: 1
  rampTo: 5

**Comment by hassy** [comment]
hi @joaormouta 👋 

that ramp definition should work in the way you want it to, although if that's not the behavior you're seeing you may be running into the bug fixed in #1683. How are you confirming the number of VUs created at the moment when you run your test?

**Comment by juanig1** [comment]
Hey Joao!

The current implementation of the `rampTo` phase uses probability to calculate arrivals. That adds a bit of noise to `arrivalRates`, especially on lower arrivals like your example:
![image](https://user-images.githubusercontent.com/102969687/212180989-b3d0d0ac-5e21-42d2-8cea-ef4fe48ff19e.png) (also it means that you can't be sure that running the same scenario twice nets you the same arrivals)

We have a new version [here](https://github.com/artilleryio/artillery/pull/1725), that calculates steps and arrivals before running and should be quite a bit more smooth:
![image](https://user-images.githubusercontent.com/102969687/212181239-81fe4732-45f6-49e4-93e2-16db6a46c102.png)

Steps are not exactly the sames as yours, because:
- it aims for exactly 150 * ((1 + 5) / 2) arrivals as it follows a linear function assuming f(0)=1 and f(150-1*)=5 (more info [here](https://github.com/artilleryio/artillery/pull/1725/files#diff-b5f0756e98806808ec6ae135920fddfaf6e7b4f0e2b9fea3807fd2bcad7d67f1R114)) 
- for the same reason, initial and final are shorter steps (1 and 5) are shorter
- arrivals are calculated on a 1s step


Would be really happy if you want to give the PR a go and/or to hear what you think!
